Flaky-test: TestDefaultMessageFormatter.testFormatMessage is flaky

TestDefaultMessageFormatter is flaky. The testFormatMessage test method fails sporadically.

After adding a debug message to the assertion `Assert.assertTrue(0 < i2, "i2 was " + i2);`, this is the exception:
```
[ERROR] testFormatMessage(org.apache.pulsar.testclient.TestDefaultMessageFormatter)  Time elapsed: 0.151 s  <<< FAILURE!
java.lang.AssertionError: i2 was 0 expected [true] but found [false]
	at org.testng.Assert.fail(Assert.java:99)
	at org.testng.Assert.failNotEquals(Assert.java:1037)
	at org.testng.Assert.assertTrue(Assert.java:45)
	at org.apache.pulsar.testclient.TestDefaultMessageFormatter.testFormatMessage(TestDefaultMessageFormatter.java:69)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:498)
	at org.testng.internal.MethodInvocationHelper.invokeMethod(MethodInvocationHelper.java:132)
	at org.testng.internal.InvokeMethodRunnable.runOne(InvokeMethodRunnable.java:45)
	at org.testng.internal.InvokeMethodRunnable.call(InvokeMethodRunnable.java:73)
	at org.testng.internal.InvokeMethodRunnable.call(InvokeMethodRunnable.java:11)
	at java.util.concurrent.FutureTask.run(FutureTask.java:266)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624)
	at java.lang.Thread.run(Thread.java:748)
```

--- a/pulsar-testclient/src/main/java/org/apache/pulsar/testclient/DefaultMessageFormatter.java
+++ b/pulsar-testclient/src/main/java/org/apache/pulsar/testclient/DefaultMessageFormatter.java
@@ -101,10 +101,14 @@ public class DefaultMessageFormatter implements IMessageFormatter {
     }
 
     private String getIntValue(float size) {
-        if (size == 0) {
-            return String.valueOf(r.nextInt());
+        int i = 0;
+        if (size != 0) {
+            i = (int) _getFloatValue(size);
         }
-        return String.valueOf((int) _getFloatValue(size));
+        if (i == 0) {
+            i = r.nextInt() + 1;
+        }
+        return String.valueOf(i);
     }
     private String getLongValue(float size) {
         if (size == 0) {

--- a/pulsar-testclient/src/test/java/org/apache/pulsar/testclient/TestDefaultMessageFormatter.java
+++ b/pulsar-testclient/src/test/java/org/apache/pulsar/testclient/TestDefaultMessageFormatter.java
@@ -66,7 +66,7 @@ public class TestDefaultMessageFormatter {
         Assert.assertTrue(l3 > 0);
         Assert.assertTrue(l3 <= 99999);
         Assert.assertTrue(i2 < 10);
-        Assert.assertTrue(0 < i2);
+        Assert.assertTrue(0 < i2, "i2 was " + i2);
         Assert.assertTrue(f2 < 100000);
         Assert.assertTrue( -100000 < f2);
 

